browser/ui/views/brave_help_bubble/brave_help_bubble_delegate_view.cc

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/ui/views/brave_help_bubble/brave_help_bubble_delegate_view.h"
+
+#include <utility>
+
+#include "base/strings/utf_string_conversions.h"
+#include "components/grit/brave_components_strings.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/metadata/metadata_impl_macros.h"
+#include "ui/gfx/canvas.h"
+#include "ui/gfx/scoped_canvas.h"
+#include "ui/views/border.h"
+#include "ui/views/bubble/bubble_border.h"
+#include "ui/views/bubble/bubble_border_arrow_utils.h"
+#include "ui/views/bubble/bubble_frame_view.h"
+#include "ui/views/controls/label.h"
+#include "ui/views/widget/widget_delegate.h"
+
+using views::BubbleBorder;
+using views::BubbleDialogDelegateView;
+using views::BubbleFrameView;
+using views::NonClientFrameView;
+
+namespace {
+constexpr SkColor kBgColor = SkColorSetARGB(0xFF, 0x20, 0x4A, 0xE3);
+
+// This class paints a border with only an arrow, skipping the painting of
+// shadow and border around the overall bounds.
+class BorderWithArrow : public views::BubbleBorder {
+ public:
+  enum BubbleArrowPart { kFill, kBorder };
+
+  explicit BorderWithArrow(Arrow arrow, ui::ColorId color_id)
+      : views::BubbleBorder(arrow,
+                            BubbleBorder::Shadow::STANDARD_SHADOW,
+                            color_id) {
+    set_visible_arrow(true);
+  }
+
+  BorderWithArrow(const BorderWithArrow&) = delete;
+  BorderWithArrow& operator=(const BorderWithArrow&) = delete;
+  ~BorderWithArrow() override = default;
+
+  // views::BubbleBorder:
+  void Paint(const views::View& view, gfx::Canvas* canvas) override {
+    PaintVisibleArrow(view, canvas);
+  }
+
+ private:
+  // We are copying this function from the upstream code because we only need to
+  // paint an arrow.
+  void PaintVisibleArrow(const views::View& view, gfx::Canvas* canvas) {
+    // The BubbleBorder subclass provides access to the visible_arrow_rect_
```

Comment:
Thinking out loud: I'd tried find out if we avoid copying the code, but no luck.